src/hotspot/cpu/aarch64/universalUpcallHandler_aarch64.cpp

```diff
@@ -71,7 +71,8 @@ address ProgrammableUpcallHandler::generate_upcall_stub(jobject rec, jobject jab
 
   // Capture prev stack pointer (stack arguments base)
   __ add(rscratch1, rfp, 16);   // Skip saved FP and LR
-  __ str(rscratch1, Address(sp, layout.stack_args));
+  Address slot = __ legitimize_address(Address(sp, layout.stack_args), wordSize, rscratch2);
```

Comment:
Introduction of legitimize_address here seems unrelated to this PR.  What's this about?

Reply:
This is a bug that Andrew pointed out. I could file a different CR for this and take it out.  It seemed a minor thing to include with this change though.

Reply:
> Introduction of legitimize_address here seems unrelated to this PR. What's this about?

SP-relative loads and stores have only a 12-bit offset. This is a bug that was discovered during review.

Reply:
Thanks for the explanation.